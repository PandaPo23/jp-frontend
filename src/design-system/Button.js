import system from '../utils/System';
import Text from './Text';

const Button = system(
  'Button',
  {
    extend: Text,
    as: 'button',
    display: 'inline-block',
    py: 1,
    px: 2,
    borderRadius: 'sm',
    minWidth: 100,
    border: 1,
    hoverable: true,
  },
  (props) => ({
    appearance: 'none',
  })
);

export default Button;
