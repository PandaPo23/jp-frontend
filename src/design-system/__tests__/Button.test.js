import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../Button';

describe('Button component', () => {
  it('renders default button correctly', () => {
    const wrapper = mount(<Button>Click Me</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders active button correctly', () => {
    const wrapper = mount(<Button active>Active</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders primary button correctly', () => {
    const wrapper = mount(<Button primary>Primary</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders secondary button correctly', () => {
    const wrapper = mount(<Button secondary>Secondary</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders info button correctly', () => {
    const wrapper = mount(<Button info>Info</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders info button correctly', () => {
    const wrapper = mount(<Button info>Info</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders error button correctly', () => {
    const wrapper = mount(<Button error>Error</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders warning button correctly', () => {
    const wrapper = mount(<Button warning>Warning</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders warning button correctly', () => {
    const wrapper = mount(<Button disabled>Disabled</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
