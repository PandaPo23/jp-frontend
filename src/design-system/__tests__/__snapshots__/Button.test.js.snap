// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders active button correctly 1`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  active={true}
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  minWidth={100}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
  >
    Active
  </button>
</Button>
`;

exports[`Button component renders default button correctly 1`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  minWidth={100}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
  >
    Click Me
  </button>
</Button>
`;

exports[`Button component renders error button correctly 1`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  error={true}
  hoverable={true}
  minWidth={100}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
  >
    Error
  </button>
</Button>
`;

exports[`Button component renders info button correctly 1`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  info={true}
  minWidth={100}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
  >
    Info
  </button>
</Button>
`;

exports[`Button component renders info button correctly 2`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  info={true}
  minWidth={100}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
  >
    Info
  </button>
</Button>
`;

exports[`Button component renders primary button correctly 1`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  minWidth={100}
  primary={true}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
  >
    Primary
  </button>
</Button>
`;

exports[`Button component renders secondary button correctly 1`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  minWidth={100}
  px={2}
  py={1}
  secondary={true}
>
  <button
    className="emotion-0"
  >
    Secondary
  </button>
</Button>
`;

exports[`Button component renders warning button correctly 1`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  minWidth={100}
  px={2}
  py={1}
  warning={true}
>
  <button
    className="emotion-0"
  >
    Warning
  </button>
</Button>
`;

exports[`Button component renders warning button correctly 2`] = `
.emotion-0 {
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<Button
  as="button"
  border={1}
  borderRadius="sm"
  disabled={true}
  display="inline-block"
  hoverable={true}
  minWidth={100}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
    disabled={true}
  >
    Disabled
  </button>
</Button>
`;
